ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -400,6 +418,36 @@ func retrieveInstanceType(cfnParams *cloudformation.CfnStackParams) (string, err
 	return aws.StringValue(param.ParameterValue), nil
 }
 
+func populateInstanceTypeParameter(cfnParams *cloudformation.CfnStackParams) (string, error) {
+	param, err := cfnParams.GetParameter(ParameterKeyInstanceType)
+	if err == cloudformation.ParameterNotFoundError {
+		logrus.Infof("Defaulting instance type to %s", cloudformation.DefaultECSInstanceType)
+
+		cfnParams.Add(ParameterKeyInstanceType, cloudformation.DefaultECSInstanceType)
+
+		return cloudformation.DefaultECSInstanceType, nil
+	} else if err != nil {
+		return "", err
+	}
+
+	return aws.StringValue(param.ParameterValue), err
```

Comment:
```suggestion
	param, err := cfnParams.GetParameter(ParameterKeyInstanceType)
	if err == nil {
	        return aws.StringValue(param.ParameterValue), err
	}
	if err == cloudformation.ParameterNotFoundError {
		logrus.Infof("Defaulting instance type to %s", cloudformation.DefaultECSInstanceType)

		cfnParams.Add(ParameterKeyInstanceType, cloudformation.DefaultECSInstanceType)

		return cloudformation.DefaultECSInstanceType, nil
	}
	return "", err
```
Maybe like this?

Reply:
It's more standard to left-align the happy path. We should check for non-nil errors first and handle those.